### PR TITLE
feat: improve delegation list header

### LINKED ISSUE #8106: [Task]: update DelegationOutputTemp
### Task description

`DelegationOutputTemp` includes `validatorAddress` that is currently returned in the response but is not in the sdk's  `DelegationOutput` interface

### LINKED ISSUE #8148: [Task]: improve delegation list header
### Task description

- change "name" column to "delegation id" which is truncated and can be copied
- rename "address" column header to "delegated to"
- `Epoch` -> `Epochs`
- update epochs logic -> current epoch - startEpoch -2 (epoch in which we started delegating and the next one do not count until it ends for rewards)

### PR DESCRIPTION
**Pull request by cpl121**
Closes #8148 
Closes #8106 

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/views/dashboard/delegation/Delegation.svelte [hunk 1]
```diff
@@ -17,51 +17,49 @@
...
-                const delegationOutput = output.output as DelegationOutputTemp
+            delegationOutputs?.map(async (output) => {
+                const delegationOutput = output.output as DelegationOutput
+                // Until the first epoch in which it was delegated ends, no rewards are obtained
+                const epochsDelegating = currentEpoch - delegationOutput.startEpoch
                 return {
-                    [Header.Name]: `Delegation ${index + 1}`,
+                    [Header.DelegationId]: delegationOutput.delegationId,
                     [Header.DelegatedFunds]: Number(delegationOutput.delegatedAmount),
                     [Header.Rewards]: await getOutputRewards(output.outputId),
```
Comment (begonaalvarezd) [inline packages/desktop/views/dashboard/delegation/Delegation.svelte:60]: `getOutputRewards` should not live in this component, i would move that logic out so its reusable somewhere else, these types of logics dont make sense in a svelte component

File: packages/desktop/views/dashboard/delegation/Delegation.svelte [hunk 2]
```diff
@@ -17,51 +17,49 @@
...
     $: ({ baseCoin } = $selectedWalletAssets[$activeProfile?.network.id])
 
     async function mappedDelegationData(delegationOutputs: OutputData[]): Promise<void> {
         const result =
-            delegationOutputs?.map(async (output, index) => {
-                const delegationOutput = output.output as DelegationOutputTemp
+            delegationOutputs?.map(async (output) => {
+                const delegationOutput = output.output as DelegationOutput
+                // Until the first epoch in which it was delegated ends, no rewards are obtained
+                const epochsDelegating = currentEpoch - delegationOutput.startEpoch
```
[the remark this answers is not in the document]
Comment (cpl121) [reply packages/desktop/views/dashboard/delegation/Delegation.svelte:56]: I also thought like this, but when a delegation output is created, the startEpoch is `currentEpoch + 1`
In the tests I did, the `currentEpoch` was 3 and the `startEpoch` of the newly created delegation output was 4

File: packages/shared/lib/core/wallet/actions/getOutputRewards.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { getClient } from './getClient'
+
+export async function getOutputRewards(outputId: string): Promise<number> {
+    const client = await getClient()
+    const rewards = await client.getOutputManaRewards(outputId)
+    return Number(rewards)
```
Comment (marc2332) [inline packages/shared/lib/core/wallet/actions/getOutputRewards.ts:6]: You are casting an object to a number :thinking: 
image

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/wallet/actions/getOutputRewards.ts (hunk 3 above)
Comment (marc2332) [reply packages/shared/lib/core/wallet/actions/getOutputRewards.ts:6]: I think you wanted to access the `rewards` property, right?

**Review by marc2332 — APPROVED** [review]
:horse_racing:

**Pull request merged**